spec/stat_tracker_spec.rb

```diff
@@ -382,4 +382,93 @@
     expect(stat_tracker.most_goals_scored('6')).to eq(3)
   end
 
+  it 'returns fewest_goals_scored' do
+    expect(stat_tracker.fewest_goals_scored('6')).to eq(2)
+  end
+
+  it 'returns favorite_opponent' do
+    fake_games = [
+      {
+        :away_team_id => '3', #6 wins vs 3
+        :home_team_id => '6',
+        :away_goals => "2",
+        :home_goals => "3"
+      },
+      {
+        :away_team_id => '6', #6 wins vs 3
+        :home_team_id => '3',
+        :away_goals => "3",
+        :home_goals => "2"
+      },
+      {
+        :away_team_id => '36', #6 loses vs 36
+        :home_team_id => '6',
+        :away_goals => "3",
+        :home_goals => "1"
+      },
+      {
+        :away_team_id => '6', #6 wins vs 36
+        :home_team_id => '36',
+        :away_goals => "4",
+        :home_goals => "3"
+      }
+    ]
+
+    fake_team_info = [
+      { :teamname => "Houston Dynamo",
+        :team_id => "3",
```

Comment:
I'm still a little confused as to why we can't include the ids in the fake_games hash